### PR TITLE
qa: Fix a few tracebacks in vstart_runner 

### PR DESCRIPTION
**Pull request by kotreshhr**
1. AttributeError: 'LocalCephManager' object has no attribute 'cluster'
2. CommandFailedError: Command failed with status 127: \
	['None/archive/coverage', 'rados' ...]
3. TypeError: a bytes-like object is required, not '_io.BytesIO'

Fixes: https://tracker.ceph.com/issues/53601
Signed-off-by: Kotresh HR <khiremat@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by kotreshhr** [comment]
Detailed tracebacks is as below

Following tracebacks are seen

1.
<pre>
2021-12-14 13:26:31,935.935 INFO:__main__:======================================================================
2021-12-14 13:26:31,935.935 INFO:__main__:ERROR: test_fragmented_injection (tasks.cephfs.test_data_scan.TestDataScan)
2021-12-14 13:26:31,936.936 INFO:__main__:That when injecting a dentry into a fragmented directory, we put it in the right fragment.
2021-12-14 13:26:31,936.936 INFO:__main__:----------------------------------------------------------------------
2021-12-14 13:26:31,936.936 INFO:__main__:Traceback (most recent call last):
2021-12-14 13:26:31,936.936 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/test_data_scan.py", line 464, in test_fragmented_injection
2021-12-14 13:26:31,936.936 INFO:__main__:    keys = self._dirfrag_keys(frag_obj_id)
2021-12-14 13:26:31,936.936 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/test_data_scan.py", line 416, in _dirfrag_keys
2021-12-14 13:26:31,936.936 INFO:__main__:    keys_str = self.fs.radosmo(["listomapkeys", object_id], stdout=StringIO())
2021-12-14 13:26:31,936.936 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/filesystem.py", line 1116, in radosmo
2021-12-14 13:26:31,936.936 INFO:__main__:    return self.radosm(*args, **kwargs, stdout=stdout).stdout.getvalue()
2021-12-14 13:26:31,936.936 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/filesystem.py", line 1109, in radosm
2021-12-14 13:26:31,936.936 INFO:__main__:    return self.rados(*args, **kwargs, pool=self.get_metadata_pool_name())
2021-12-14 13:26:31,937.937 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/filesystem.py", line 1102, in rados
2021-12-14 13:26:31,937.937 INFO:__main__:    return self.mon_manager.do_rados(*args, **kwargs)
2021-12-14 13:26:31,937.937 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/ceph_manager.py", line 1728, in do_rados
2021-12-14 13:26:31,937.937 INFO:__main__:    self.cluster,
2021-12-14 13:26:31,937.937 INFO:__main__:AttributeError: 'LocalCephManager' object has no attribute 'cluster'
</pre>

2.

<pre>
2021-12-14 13:33:01,613.613 INFO:__main__:ERROR: test_fragmented_injection (tasks.cephfs.test_data_scan.TestDataScan)
2021-12-14 13:33:01,614.614 INFO:__main__:That when injecting a dentry into a fragmented directory, we put it in the right fragment.
2021-12-14 13:33:01,614.614 INFO:__main__:----------------------------------------------------------------------
2021-12-14 13:33:01,614.614 INFO:__main__:Traceback (most recent call last):
2021-12-14 13:33:01,614.614 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/test_data_scan.py", line 464, in test_fragmented_injection
2021-12-14 13:33:01,614.614 INFO:__main__:    keys = self._dirfrag_keys(frag_obj_id)
2021-12-14 13:33:01,614.614 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/test_data_scan.py", line 416, in _dirfrag_keys
2021-12-14 13:33:01,614.614 INFO:__main__:    keys_str = self.fs.radosmo(["listomapkeys", object_id], stdout=StringIO())
2021-12-14 13:33:01,614.614 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/filesystem.py", line 1116, in radosmo
2021-12-14 13:33:01,614.614 INFO:__main__:    return self.radosm(*args, **kwargs, stdout=stdout).stdout.getvalue()
2021-12-14 13:33:01,614.614 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/filesystem.py", line 1109, in radosm
2021-12-14 13:33:01,614.614 INFO:__main__:    return self.rados(*args, **kwargs, pool=self.get_metadata_pool_name())
2021-12-14 13:33:01,614.614 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/filesystem.py", line 1102, in rados
2021-12-14 13:33:01,614.614 INFO:__main__:    return self.mon_manager.do_rados(*args, **kwargs)
2021-12-14 13:33:01,615.615 INFO:__main__:  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/ceph_manager.py", line 1735, in do_rados
2021-12-14 13:33:01,615.615 INFO:__main__:    proc = remote.run(
2021-12-14 13:33:01,615.615 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 410, in run
2021-12-14 13:33:01,615.615 INFO:__main__:    return self._do_run(**kwargs)
2021-12-14 13:33:01,615.615 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 480, in _do_run
2021-12-14 13:33:01,615.615 INFO:__main__:    proc.wait()
2021-12-14 13:33:01,615.615 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 221, in wait
2021-12-14 13:33:01,615.615 INFO:__main__:    raise CommandFailedError(self.args, self.exitstatus)
2021-12-14 13:33:01,615.615 INFO:__main__:teuthology.exceptions.CommandFailedError: Command failed with status 127: ['None/archive/coverage', 'rados', '--cluster', 'ceph', '--pool', 'cephfs_metadata', 'listomapkeys', '10000000000.01000000']
</pre>

3.
<pre>
2021-12-14 15:13:34,092.092 ERROR:__main__:Traceback (most recent call last):
  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/test_data_scan.py", line 580, in test_rebuild_linkage
    self.fs.radosm(["setomapval", dirfrag2_oid, file1_key], stdin=BytesIO(file1_omap_data))
  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/filesystem.py", line 1109, in radosm
    return self.rados(*args, **kwargs, pool=self.get_metadata_pool_name())
  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/cephfs/filesystem.py", line 1102, in rados
    return self.mon_manager.do_rados(*args, **kwargs)
  File "/root/sandbox/kotresh-ceph/ceph/qa/tasks/ceph_manager.py", line 1735, in do_rados
    proc = remote.run(
  File "../qa/tasks/vstart_runner.py", line 410, in run
    return self._do_run(**kwargs)
  File "../qa/tasks/vstart_runner.py", line 472, in _do_run
    subproc.stdin.write(stdin)
TypeError: a bytes-like object is required, not '_io.BytesIO'
</pre>

**Comment by kotreshhr** [comment]
jenkins test needs-rebase

File: qa/tasks/vstart_runner.py [hunk 1]
```diff
@@ -423,7 +423,7 @@ def _do_run(self, args, check_status=True, wait=True, stdout=None,
 
         # Filter out helper tools that don't exist in a vstart environment
         args = [a for a in args if a not in ('adjust-ulimits',
-                                             'ceph-coverage')]
+                                             'ceph-coverage', 'None/archive/coverage')]
```
Comment (lxbsz) [inline qa/tasks/vstart_runner.py:426]: I think it's from here:

https://github.com/ceph/ceph/blob/master/qa/tasks/ceph_manager.py#L1725

Maybe we should filter out all the `*/archive/coverage` ?

Comment (kotreshhr) [reply qa/tasks/vstart_runner.py:426]: Not required. It's always set to `None` in vstart_runner https://github.com/ceph/ceph/pull/44305/files/d01ef7682c1b1a9a6de28d458d64832708b0b3e2#R797

**Review by lxbsz — APPROVED** [review]

File: qa/tasks/vstart_runner.py (hunk 1 above)
Comment (lxbsz) [reply qa/tasks/vstart_runner.py:426]: Okay.

**Review by rishabh-d-dave — COMMENTED** [review]
I couldn't run the test successfully even after applying this PR. `run_shell()` needed a correction in order to run the test successfully. I've pushed that patch too on this branch.

**Comment by rishabh-d-dave** [comment]
@kotreshhr My attempt to push commits to your PR branch wasn't successful. I don't know what's wrong. I've posted links to my commits in respective comments now.

File: qa/tasks/vstart_runner.py [hunk 2]
```diff
@@ -808,6 +808,8 @@ def __init__(self):
         # since it doesn't accept "shell" as parameter.
         self.run_ceph_w_prefix = ['exec', 'sudo', CEPH_CMD]
 
+        self.cluster = 'ceph'
+
```
[the remark this answers is not in the document]
Comment (rishabh-d-dave) [reply qa/tasks/vstart_runner.py:812]: https://github.com/rishabh-d-dave/ceph/commit/0a005eea79d7904295aac5178dbba4f35d22d5ff

[the remark this answers is not in the document]
Comment (kotreshhr) [reply qa/tasks/vstart_runner.py]: I have pushed your commit on top of mine. @rishabh-d-dave PTAL. Your commit looks good to me.

[the remark this answers is not in the document]
Comment (rishabh-d-dave) [reply qa/tasks/vstart_runner.py]: Thanks!

**Comment by rishabh-d-dave** [comment]
> Link to my patch for run_shell() modification - rishabh-d-dave@8f82c6e

@kotreshhr This commit is on the same branch. The test failed for me without this patch. So, will you please push that patch too on this branch?

**Comment by kotreshhr** [comment]
> @kotreshhr My attempt to push commits to your PR branch wasn't successful. I don't know what's wrong. I've posted links to my commits in respective comments now.

Done

**Comment by kotreshhr** [comment]
jenkins test make check

File: qa/tasks/cephfs/mount.py [hunk 3]
```diff
@@ -695,7 +695,7 @@ def run_python(self, pyscript, py_version='python3', sudo=False):
         p.wait()
         return p.stdout.getvalue().strip()
 
-    def run_shell(self, args, timeout=900, **kwargs):
+    def run_shell(self, args, timeout=300, **kwargs):
```
Comment (rishabh-d-dave) [inline qa/tasks/cephfs/mount.py:698]: @cephfs @batrick This change is not strictly related to this PR/commit. Is it okay to keep it here? I often am told to create a separate PR but this change is very very minor.

Comment (batrick) [reply qa/tasks/cephfs/mount.py:698]: Why is "300" more reasonable than "900"? What problem is being solved?

Comment (batrick) [reply qa/tasks/cephfs/mount.py:698]: (yes, a separate PR is better)

Comment (kotreshhr) [reply qa/tasks/cephfs/mount.py]: Done

**Comment by rishabh-d-dave** [comment]
Tested this PR a couple of times, today and yesterday. `test_data_scan.TestDataScan.test_fragmented_injection` ran successfully for me.

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
Ran  the smallest testsuite to exercise the change in mount.py, the testsuite ran successfully -

http://pulpito.front.sepia.ceph.com/rishabh-2022-02-09_17:33:56-fs-wip-vshankar-testing-20220201-113815-distro-basic-smithi/

@vshankar IMHO this PR's ready for final review and merge.

**Comment by vshankar** [comment]
> Ran the smallest testsuite to exercise the change in mount.py, the testsuite ran successfully -
> 
> http://pulpito.front.sepia.ceph.com/rishabh-2022-02-09_17:33:56-fs-wip-vshankar-testing-20220201-113815-distro-basic-smithi/
> 
> @vshankar IMHO this PR's ready for final review and merge.

Sure. I'll take a look.

**Comment by lxbsz** [comment]
@kotreshhr @rishabh-d-dave Please remove the following comments from the commit comments https://github.com/ceph/ceph/pull/44305/commits/71a6cb8a70ff856df82ada2e8ebc80cff88bf76d, which makes no sense any more:

```
Also, since we are here, let's change the timeout time to something more
reasonable.
```

**Comment by rishabh-d-dave** [comment]
> @kotreshhr @rishabh-d-dave Please remove the following comments from the commit comments [71a6cb8](https://github.com/ceph/ceph/commit/71a6cb8a70ff856df82ada2e8ebc80cff88bf76d), which makes no sense any more:
> 
> ```
> Also, since we are here, let's change the timeout time to something more
> reasonable.
> ```
> 
> All the others look good to me.

Agreed. That part of the patch is deleted now.

@kotreshhr Can you please remove that line from my commit message? I've removed it locally but I can't update the PR branch, perhaps I'm not allowed. Alternately, if you prefer so, I can push the new commit on a separate branch and you can cherry-pick the commit then.

**Comment by kotreshhr** [comment]
> > @kotreshhr @rishabh-d-dave Please remove the following comments from the commit comments [71a6cb8](https://github.com/ceph/ceph/commit/71a6cb8a70ff856df82ada2e8ebc80cff88bf76d), which makes no sense any more:
> > ```
> > Also, since we are here, let's change the timeout time to something more
> > reasonable.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > All the others look good to me.
> 
> Agreed. That part of the patch is deleted now.
> 
> @kotreshhr Can you please remove that line from my commit message? I've removed it locally but I can't update the PR branch, perhaps I'm not allowed. Alternately, if you prefer so, I can push the new commit on a separate branch and you can cherry-pick the commit then.

Done.

**Comment by rishabh-d-dave** [comment]
Not sure if make check failure is related - https://jenkins.ceph.com/job/ceph-pull-requests/94079/

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by kotreshhr** [comment]
jenkins test make check

**Comment by kotreshhr** [comment]
Rebased

**Comment by rishabh-d-dave** [comment]
@kotreshhr This PR too is going through the same issue: CI dashboard shows that readthedocs CI job is stuck but clicking the link shows that it failed for an unrelated reason. See https://readthedocs.org/projects/ceph/builds/16746721/. This PR needs to re-trigger it. There's no command that will re-trigger it, so you will have to re-push the branch to do it.

cc: @vshankar

**Comment by kotreshhr** [comment]
@vshankar PTAL

**Comment by rishabh-d-dave** [comment]
@vshankar CI is green now.

**Comment by vshankar** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2022-Apr-25
